test/e2e/runner/setup.go

```diff
@@ -166,19 +166,33 @@ func MakeConfig(node *e2e.Node) (*config.Config, error) {
 	cfg := config.DefaultConfig()
 	cfg.Moniker = node.Name
 	cfg.ProxyApp = AppAddressTCP
+
 	cfg.RPC.ListenAddress = "tcp://0.0.0.0:26657"
+	cfg.RPC.PprofListenAddress = ":6060"
+
 	cfg.GRPC.ListenAddress = "tcp://0.0.0.0:26670"
 	cfg.GRPC.VersionService.Enabled = true
-	cfg.GRPC.Privileged.ListenAddress = "tcp://0.0.0.0:26671"
-	cfg.GRPC.Privileged.PruningService.Enabled = true
-	cfg.RPC.PprofListenAddress = ":6060"
+	cfg.GRPC.BlockService.Enabled = true
+	cfg.GRPC.BlockResultsService.Enabled = true
+
 	cfg.P2P.ExternalAddress = fmt.Sprintf("tcp://%v", node.AddressP2P(false))
 	cfg.P2P.AddrBookStrict = false
+
 	cfg.DBBackend = node.Database
 	cfg.StateSync.DiscoveryTime = 5 * time.Second
 	cfg.BlockSync.Version = node.BlockSyncVersion
 	cfg.Consensus.PeerGossipIntraloopSleepDuration = node.Testnet.PeerGossipIntraloopSleepDuration
 
+	// Assume that full nodes and validators will have a data companion
+	// attached, which will need access to the privileged gRPC endpoint.
+	if (node.Mode == e2e.ModeValidator || node.Mode == e2e.ModeFull) && node.EnableCompanionPruning {
+		cfg.Storage.Pruning.DataCompanion.Enabled = true
+		cfg.Storage.Pruning.DataCompanion.InitialBlockRetainHeight = 0
```

Comment:
This means that nothing will ever be pruned right?

Reply:
Yes. Unless the data companion explicitly sets a different retain height. If I'd left these values out, they would have been initialized to 0 anyways as per Go's conventions - I was just making it explicit here.